main.tex

```diff
@@ -39,7 +54,58 @@ \section{Introduction}
 
-\section{Our Contribution}
+\subsection{Limitations of ZeroLink}
+In this work, we are aiming to improve ZeroLink, the most popular Chaumian CoinJoin implementation for Bitcoin. We identify several privacy shortcomings and inefficiencies of the current protocol as implemented by Wasabi.
+
+\subsubsection{Round denominations}
+
+
+Due to the nature of blind signatures, mixed outputs of ZeroLink CoinJoins are restricted to a fixed set of value denominations which are multiples of a base denomination\footnote{approximately $0.1$\bitcoinA{}}.
+
+This creates friction when sending or receiving arbitrary amounts of Bitcoin, as the fixed denomination generally creates change which is smaller, both when mixing and when spending mixed outputs.
+
+We define \emph{CoinJoin inefficiency} as the fraction of non-mixed outputs in a CoinJoin transaction, see Figure~\ref{fig:cjinefficiency}.
+
+\begin{figure}[h!]
+    \centering
+    \includegraphics[scale=0.4]{Figures/CJInefficiency.pdf}
+    \caption{CoinJoin inefficiency of various privacy-focused Bitcoin wallets}
+    \label{fig:cjinefficiency}
+\end{figure}
+
+\subsubsection{Minimum denomination}
+
+In order to pariticpate a user's combined input amount are must be greater or equal to the base denomination. We observe, that considerable portion of CoinJoin inputs are less than this minimum denomination, see Figure~\ref{fig:minimumdenomination}.
+
+\begin{figure}[h!]
+    \centering
+    \includegraphics[scale=0.4]{Figures/SmallValueInputsWasabi.pdf}
+    \caption{Fraction of input UTXOs with value smaller than the minimum denomination in Wasabi CoinJoin transactions}
+    \label{fig:minimumdenomination}
+\end{figure}
+
+Even if users are able to provide several smaller value input UTXOs with total value greater than the minimum denomination, the coordinator knows those input UTXOs belong to the same user.  In an ideal mixing protocol the coordinator should not obtain more information than the already available blockchain data by coordinating the CoinJoin transaction.
+
+Furthermore if users consolidate their UTXOs before the CoinJoin in an additional transaction in order to be able to participate in a CoinJoin, then this link is revealed publicly based on the common ownership heuristic~\cite{meiklejohn2013fistful}.
+
+\subsubsection{Variable denominations} Since users pay mining and coordination fees the denominations are gradually reduced between rounds in order to make it possible for users to mix several times without providing additional inputs. This introduces a perverse incentive to minimize coordination fees by remixing in quick succession in order, resulting in a smaller anonymity set % TODO how to say - because users are more "clumped together" in time
```

Comment:
```suggestion
\subsubsection{Variable denominations} Since users pay mining and coordination fees the denominations are gradually reduced between rounds in order to make it possible for users to mix several times without providing additional inputs. This introduces a perverse incentive to minimize coordination fees by remixing in quick succession in order, resulting in a smaller anonymity set compared to time-staggered remixes.
```